client/active-game/game-server.js

```diff
@@ -41,15 +41,28 @@ class GameServer {
         log.verbose('game websocket connected')
         socket.on('close', () => {
           log.verbose('game websocket disconnected')
+          const socket = this.idToSocket.get(gameId)
+          if (socket) {
+            clearInterval(socket.pingInterval)
+          }
           this.idToSocket = this.idToSocket.delete(gameId)
         })
         socket.on('message', message => {
           this.onMessage(gameId, message)
         })
-        this.idToSocket = this.idToSocket.set(gameId, socket)
+        socket.on('error', e => {
+          log.error(`Game socket error ${e}`)
+        })
+        const pingInterval = setInterval(() => {
+          socket.ping()
+        }, 20000)
+        this.idToSocket = this.idToSocket.set(gameId, { socket, pingInterval })
```

Comment:
I'd prefer to use two different maps for this; or at least give a better name for this new, combined map. Which would hopefully eliminate the usage of `socket.socket` above which looks pretty weird and confusing ^^
Also, on line 44 you're shadowing the `socket` variable as well. Either using two different maps or using a differently named map should fix those issues.